src/azure-cli/azure/cli/command_modules/batch/_params.py

```diff
@@ -71,15 +72,44 @@ def load_arguments(self, _):
-        c.argument('identity_type', help="The type of identity used for the Batch account. Possible values include: 'SystemAssigned', 'None'.", arg_type=get_enum_type(ResourceIdentityType))
+        c.argument('identity_type', help="The type of identity used for the Batch account.", arg_type=get_enum_type(ResourceIdentityType))
+        c.argument('mi_user_assigned', help="The managed User identity fully qualified resource Id. Must be in conjunction with identity_type=UserAssigned")
         c.ignore('keyvault_url')
 
     with self.argument_context('batch account set') as c:
         c.argument('tags', tags_type)
         c.argument('storage_account', help='The storage account name or resource ID to be used for auto storage.', validator=storage_account_id)
         c.argument('encryption_key_source', help='Part of the encryption configuration for the Batch account. Type of the key source. Can be either Microsoft.Batch or Microsoft.KeyVault')
+        c.argument('public_network_access', help="The network access type for accessing Azure Batch account. Values can either be enabled or disabled.", arg_type=get_enum_type(PublicNetworkAccessType))
         c.argument('encryption_key_identifier', help='Part of the encryption configuration for the Batch account. Full path to the versioned secret. Example https://mykeyvault.vault.azure.net/keys/testkey/6e34a81fef704045975661e297a4c053.')
-        c.argument('identity_type', help="The type of identity used for the Batch account. Possible values include: 'SystemAssigned', 'None'.", arg_type=get_enum_type(ResourceIdentityType))
+        c.argument('identity_type', help="The type of identity used for the Batch account.", arg_type=get_enum_type(ResourceIdentityType))
+        c.argument('mi_user_assigned', help="The managed User identity fully qualified resource Id. Must be in conjunction with identity_type=UserAssigned")
+
+    with self.argument_context('batch account network-profile show') as c:
+        c.argument('resource_group_name', resource_group_name_type, help='Name of the resource group. If not specified will display currently set account.', required=False)
+        c.argument('account_name', batch_name_type, options_list=('--name', '-n'), help='Name of the batch account to show. If not specified will display currently set account.', required=False)
+
+    with self.argument_context('batch account network-profile set') as c:
+        c.argument('resource_group_name', resource_group_name_type, help='Name of the resource group. If not specified will display currently set account.', required=False)
+        c.argument('account_name', batch_name_type, options_list=('--name', '-n'), help='Name of the batch account to show. If not specified will display currently set account.', required=False)
+        c.argument('profile', help="Network profile to set. Allowed values include: 'BatchAccount', 'NodeManagement'")
```

Comment:
Could we define enumerations for these valid values?

Reply:
enum defined